core/src/main/scala/org/broadinstitute/dsde/rawls/dataaccess/HttpGoogleServicesDAO.scala

```diff
@@ -620,6 +620,45 @@ class HttpGoogleServicesDAO(
     }
   }
 
+  override def updateGoogleProjectBillingAccount(googleProjectId: GoogleProjectId, billingAccount: Option[RawlsBillingAccountName]): Future[ProjectBillingInfo] = {
+    val billingSvcCred = getBillingServiceAccountCredential
+    implicit val service = GoogleInstrumentedService.Billing
+    val googleProjectName = s"projects/${googleProjectId.value}"
+    val cloudBillingProjectsApi = getCloudBillingManager(billingSvcCred).projects()
+
+    val fetcher = cloudBillingProjectsApi.getBillingInfo(googleProjectName)
+
+    val updater = billingAccount match {
+      case Some(RawlsBillingAccountName(billingAccountName)) =>
+        cloudBillingProjectsApi.updateBillingInfo(googleProjectName,
+          new ProjectBillingInfo().setBillingAccountName(billingAccountName).setBillingEnabled(true))
+      case None =>
+        cloudBillingProjectsApi.updateBillingInfo(googleProjectName,
+          new ProjectBillingInfo().setBillingEnabled(false))
+    }
+    retryWithRecoverWhen500orGoogleError(() => {
+      blocking {
+        val projectBillingInfo = executeGoogleRequest(fetcher)
+        val shouldUpdate = billingAccount match {
+          case Some(RawlsBillingAccountName(billingAccountName)) =>
+            projectBillingInfo.getBillingAccountName == billingAccountName && projectBillingInfo.getBillingEnabled == true
+          case None =>
+            projectBillingInfo.getBillingEnabled == false
+        }
```

Comment:
this logic looks backwards, `shouldUpdate` should be true when `projectBillingInfo` differs from what is intended by `billingAccount`

Also we probably should be careful with possible nulls coming back in `projectBillingInfo`, e.g.  `Option(projectBillingInfo.getBillingAccountName).contains(billingAccountName)`